apidef/oas/oas.go

```diff
@@ -23,6 +23,19 @@ func (s *OAS) MarshalJSON() ([]byte, error) {
 		s.Info.License = nil
 	}
 
+	// when OAS object is unmarshalled, the extension values are marshalled as plain []byte by kin/openapi
+	// this causes json marshaller to base64 encode the values - https://pkg.go.dev/encoding/json#Marshal.
+	// this block converts the extensions to json.RawMessage so that it's correctly marshalled.
+	for k := range s.Extensions {
```

Comment:
I wonder whether there is an internal function of the library itself

Reply:
https://github.com/getkin/kin-openapi/blob/master/jsoninfo/marshal.go#L37
The `[]byte` is set to the extension keys here, if it's set to `json.RawMessage` from there, these changes can be removed. We can file an issue on upstream.